hadoop-common-project/hadoop-common/src/test/java/org/apache/hadoop/security/TestSaslPropertiesResolver.java

```diff
@@ -0,0 +1,49 @@
+package org.apache.hadoop.security;
+
+import java.net.InetAddress;
+import java.net.InetSocketAddress;
+import javax.security.sasl.Sasl;
+
+import org.junit.Before;
+import org.junit.Test;
+
+import org.apache.hadoop.conf.Configuration;
+
+import static org.apache.hadoop.fs.CommonConfigurationKeysPublic.HADOOP_RPC_PROTECTION;
+import static org.junit.Assert.assertEquals;
+
+public class TestSaslPropertiesResolver {
+
+  private static final SaslRpcServer.QualityOfProtection PRIVACY_QOP = SaslRpcServer.QualityOfProtection.PRIVACY;
+  private static final SaslRpcServer.QualityOfProtection AUTHENTICATION_QOP = SaslRpcServer.QualityOfProtection.AUTHENTICATION;
+  private static final InetAddress LOCALHOST = new InetSocketAddress("127.0.0.1", 1).getAddress();
+
+  private SaslPropertiesResolver resolver;
+
+  @Before
+  public void setup() {
+    Configuration conf = new Configuration();
+    conf.set(HADOOP_RPC_PROTECTION, "privacy");
+    resolver = new SaslPropertiesResolver();
+    resolver.setConf(conf);
+  }
+
+  @Test
+  public void testResolverDoesNotMutate() {
+    assertEquals(PRIVACY_QOP.getSaslQop(), resolver.getDefaultProperties().get(Sasl.QOP));
```

Comment:
lot of duplication here; 
is it possible to factor these out into new assertions? at the very least some assertPrivacyQop() which is called all the way down.

Reply:
Removed a bunch of duplication.